src/Domain/DomainCreators/Disk.cpp

```diff
@@ -0,0 +1,103 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#include "Domain/DomainCreators/Disk.hpp"
+
+#include <cmath>
+#include <vector>
+
+#include "Domain/Block.hpp"
+#include "Domain/BlockNeighbor.hpp"
+#include "Domain/CoordinateMaps/AffineMap.hpp"
+#include "Domain/CoordinateMaps/Equiangular.hpp"
+#include "Domain/CoordinateMaps/ProductMaps.hpp"
+#include "Domain/CoordinateMaps/Wedge2D.hpp"
+#include "Domain/Direction.hpp"
+#include "Domain/Domain.hpp"
+#include "Utilities/MakeVector.hpp"
+
+namespace DomainCreators {
+
+Disk::Disk(typename InnerRadius::type inner_radius,
+           typename OuterRadius::type outer_radius,
+           typename InitialRefinement::type initial_refinement,
+           typename InitialGridPoints::type initial_number_of_grid_points,
+           typename UseEquiangularMap::type use_equiangular_map) noexcept
+    // clang-tidy: trivially copyable
+    : inner_radius_(std::move(inner_radius)),         // NOLINT
+      outer_radius_(std::move(outer_radius)),         // NOLINT
+      initial_refinement_(                            // NOLINT
+          std::move(initial_refinement)),             // NOLINT
+      initial_number_of_grid_points_(                 // NOLINT
+          std::move(initial_number_of_grid_points)),  // NOLINT
+      use_equiangular_map_(use_equiangular_map) {}    // NOLINT
+
+Domain<2, Frame::Inertial> Disk::create_domain() const noexcept {
+  using Wedge2DMap = CoordinateMaps::Wedge2D;
+  using Wedge2DType =
+      CoordinateMap<Frame::Logical, Frame::Inertial, Wedge2DMap>;
+  using AffineMap = CoordinateMaps::AffineMap;
+  using AffineMap2D = CoordinateMaps::ProductOf2Maps<AffineMap, AffineMap>;
+  using Equiangular = CoordinateMaps::Equiangular;
+  using Equiangular2D =
+      CoordinateMaps::ProductOf2Maps<Equiangular, Equiangular>;
+
+  std::array<size_t, 4> block0_corners{{1, 5, 3, 7}},  //+x wedge
+      block1_corners{{3, 7, 2, 6}},                    //+y wedge
+      block2_corners{{2, 6, 0, 4}},                    //-x wedge
+      block3_corners{{0, 4, 1, 5}},                    //-y wedge
+      block4_corners{{0, 1, 2, 3}};                    // Center square
+
+  std::vector<std::array<size_t, 4>> corners{block0_corners, block1_corners,
+                                             block2_corners, block3_corners,
+                                             block4_corners};
+  if (use_equiangular_map_) {
```

Comment:
I think all this can be shortened to:

```cpp
auto coord_maps = make_vector_coordinate_map_base<Frame::Logical, TargetFrame>(
    Wedge2DMap{inner_radius_, outer_radius_, Direction<2>::upper_xi(),
               use_equiangular_map_},
    Wedge2DMap{inner_radius_, outer_radius_, Direction<2>::upper_eta(),
               use_equiangular_map_},
    Wedge2DMap{inner_radius_, outer_radius_, Direction<2>::lower_xi(),
               use_equiangular_map_},
    Wedge2DMap{inner_radius_, outer_radius_, Direction<2>::lower_eta(),
               use_equiangular_map_});

if (use_equiangular_map_) {
  coord_maps.emplace_back(make_coordinate_map_base<Frame::Logical, TargetFrame>(
      Equiangular2D{Equiangular(-1.0, 1.0, -1.0 * inner_radius_ / sqrt(2.0),
                                inner_radius_ / sqrt(2.0)),
                    Equiangular(-1.0, 1.0, -1.0 * inner_radius_ / sqrt(2.0),
                                inner_radius_ / sqrt(2.0))}));
} else {
  coord_maps.emplace_back(make_coordinate_map_base<Frame::Logical, TargetFrame>(
      AffineMap2D{AffineMap(-1.0, 1.0, -1.0 * inner_radius_ / sqrt(2.0),
                            inner_radius_ / sqrt(2.0)),
                  AffineMap(-1.0, 1.0, -1.0 * inner_radius_ / sqrt(2.0),
                            inner_radius_ / sqrt(2.0))}));
}
return Domain<2, TargetFrame>{std::move(coord_maps), std::move(corners)};
```

I replaced `Frame::Inertial` because I think in general we want DomainCreators to go from Logical -> Grid, not Inertial. For now just a type alias to that would be fine, in the future we'll need to figure out how to best deal with this.